### DISCUSSION TITLE
Override highlight groups?

### DISCUSSION THREAD

**Discussion by olimorris** [mason-org/mason.nvim#133] [Q&A]
We have highlight groups in `lua/mason/ui/palette.lua` but they seem like a pain to override. Is there currently any way to do so or should I raise a feature request?

**Comment by williamboman** [comment]
Hello! I'm personally a fan of customizing highlights the good old way via the `:highlight` DSL. Is this not working for you, for example `hi MasonHeader guibg=red`?

**Reply by olimorris** [reply]
I'm a big fan of that too and it's how most of us theme authors create the highlights. I've assumed it's down to the ordering of the plugins loaded, but it seems hard to make the highlights "stick" with mason. I notice a lot of plugin authors allow for some api like `highlights = false` which seems to stop the setting of anything like the hl groups in `lua/mason/ui/palette.lua`.

As an aside...switched to mason last night and config has been a breeze! Bravo 👏🏼 .

**Reply by williamboman** [reply]
Ah, I now realize that highlight groups are only first defined when you open the UI window, at which point it'll override any existing highlights. Will have to rethink that..

> As an aside...switched to mason last night and config has been a breeze! Bravo 👏🏼 .

Cool, nice to hear!

**Reply by siduck** [reply]
@olimorris can confirm this issue :(

**Reply by williamboman** [reply]
For the time being you can probably work around this by setting highlight overrides in a `FileType mason.nvim` autocmd

**Reply by siduck** [reply]
I can wait :raised_hands:

**Reply by luetage** [reply]
> For the time being you can probably work around this by setting highlight overrides in a `FileType mason.nvim` autocmd

That’s good to know, didn’t know that the plugin had a filetype. One other thing I noticed is that without highlight modification Mason will lose its default highlights when first loading the interface and then changing the colorscheme and opening up Mason again. Presumably because the highlights have been overwritten and Mason only sets them on first load.

**Reply by luetage** [reply]
For anyone looking for an example how to do this ☛

```lua
-- mason colors
local function sethi()
  local mbg = ("#%06x"):format(vim.api.nvim_get_hl_by_name("NormalFloat", true).background)
  local yellow = vim.g.terminal_color_3
  local blue = vim.g.terminal_color_4
  local muted = ("#%06x"):format(vim.api.nvim_get_hl_by_name("Comment", true).foreground)

  vim.api.nvim_set_hl(0, "MasonHighlight", {
    fg = blue,
  })
  vim.api.nvim_set_hl(0, "MasonHighlightBlock", {
    bg = blue,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonHighlightBlockBold", {
    bold = true,
    bg = blue,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonHighlightSecondary", {
    fg = yellow,
  })
  vim.api.nvim_set_hl(0, "MasonHighlightBlockSecondary", {
    bg = yellow,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonHighlightBlockBoldSecondary", {
    bold = true,
    bg = yellow,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonLink", {
    link = "MasonHighlight",
  })
  vim.api.nvim_set_hl(0, "MasonMuted", {
    fg = muted,
  })
  vim.api.nvim_set_hl(0, "MasonMutedBlock", {
    bg = muted,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonMutedBlockBold", {
    bold = true,
    bg = muted,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonHeading", {
    bold = true,
  })
  vim.api.nvim_set_hl(0, "MasonHeader", {
    bold = true,
    bg = yellow,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonHeaderSecondary", {
    bold = true,
    bg = blue,
    fg = mbg,
  })
  vim.api.nvim_set_hl(0, "MasonError", {
    link = "Error",
  })
end

vim.api.nvim_create_augroup("_mason", { clear = true })
vim.api.nvim_create_autocmd("Filetype", {
  pattern = "mason.nvim",
  callback = function()
    sethi()
  end,
  group = "_mason",
})
```

**Reply by williamboman** [reply]
Heads up, `filetype` was changed: https://github.com/williamboman/mason.nvim/pull/317

**Reply by luetage** [reply]
> Heads up, `filetype` was changed: #317

That’s funny. Just today I put that code in a filetype plugin and noticed the dot works as separator to distinguish filetypes. Still was workable because I used `mason` alone, but thanks for the fix!
